packages/solidjs/package.json

```diff
@@ -54,6 +55,9 @@
     "solid-js": "1.8.11",
     "vite-plugin-solid": "^2.8.2"
   },
+  "optionalDependencies": {
```

Comment:
Is `optionalDependencies` the best thing for this? What about making this an optional peer dependency? (not sure how well supported etc. this is though...)

Because I guess `optionalDependencies` means this will generally always be installed along by the SDK, right...?

Reply:
I also realized this. I thought optionalDependencies would be just like a "soft" peer dep but package managers will actually try to install these. We should simply document the version requirements instead in JS Doc and in the actual docs.

Reply:
Optional peer dependencies are a npm 7.x thing. Not sure if that's too high for us.

As for `optionalDependencies` - I've only tried with a tarball, but solid router was not pulled in if the user didn't already have it and there were no warnings about requiring it either. Or am I misunderstanding your question?

Reply:
I mean maybe it is fine to have it as `optionalDependencies`, just feels like a non-ideal solution 😬 I guess this is why in react we do not depend on react-router at all 🤔

Reply:
We only use types from solid router, I could vendor these in, wdyt?

Reply:
yeah, I think that's probably the best solution here! We also do that in other places (it has it's own set of problems obviously, but I think this is the best trade off 🤔 )

Reply:
Please only add dependencies as `devDependencies`. Everything else only has downsides and no upsides (that I am aware of).

Reply:
I vendored the types in and removed the dep on solid router.